src/cleanfigure.m

```diff
@@ -743,89 +750,200 @@ function movePointsCloser(meta, handle)
   xNew = x + bsxfun(@times ,minAlpha, (xRef-x));
 end
 % =========================================================================
-function minAlpha = updateAlpha(X1, X2, X3, X4, minAlpha)
-  % Checks whether the segments X1--X2 and X3--X4 intersect. 
-  lambda = crossLines(X1, X2, X3, X4);
-  
-  % Check if lambda is in bounds and lambda1 large enough
-  id_Alpha           = 0.0 < lambda(:,2) & lambda(:,2) < 1.0 ...
-                     & abs(minAlpha) > abs(lambda(:,1));
+function [xData, yData] = getVisualData(meta, handle)
+    % Extract the data from the current line handle.
+    xData = get(handle, 'XData');
+    yData = get(handle, 'YData');
 
-  % Update alpha when applicable
-  minAlpha(id_Alpha) = lambda(id_Alpha,1);
+    % Get info about log scaling
+    isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+    if isXlog
+        xData = log10(xData);
+    end
+    isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+    if isYlog
+        yData = log10(yData);
+    end
+
+    % Turn the data into a row vector
+    xData = xData(:);
+    yData = yData(:);
 end
 % =========================================================================
-function out = isInBox(data, xLim, yLim)
+function [xLim, yLim] = getVisualLimits(meta)
+    % Get the axis limits
+    xLim = get(meta.gca, 'XLim');
+    yLim = get(meta.gca, 'YLim');
 
-  out = data(:,1) > xLim(1) & data(:,1) < xLim(2) ...
-      & data(:,2) > yLim(1) & data(:,2) < yLim(2);
+    % Check for logarithmic scales
+    isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+    if isXlog
+        xLim  = log10(xLim);
+    end
+    isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+	if isYlog
+        yLim  = log10(yLim);
+	end
 end
 % =========================================================================
-function lambda = crossLines(X1, X2, X3, X4)
-  % Checks whether the segments X1--X2 and X3--X4 intersect.
-  % See https://en.wikipedia.org/wiki/Line-line_intersection for reference.
-  % Given four points X_k=(x_k,y_k), k\in{1,2,3,4}, and the two lines 
-  % defined by those,
-  %
-  %  L1(lambda) = X1 + lambda (X2 - X1)
-  %  L2(lambda) = X3 + lambda (X4 - X3)
-  %
-  % returns the lambda for which they intersect (and Inf if they are parallel).
-  % Technically, one needs to solve the 2x2 equation system
-  %
-  %   x1 + lambda1 (x2-x1)  =  x3 + lambda2 (x4-x3)
-  %   y1 + lambda1 (y2-y1)  =  y3 + lambda2 (y4-y3)
-  %
-  % for lambda1 and lambda2.
+function replaceData(handle, id_replace, dataReplace)
+  % Replaces data at id_replace with dataReplace
```

Comment:
As a matter of fact, we only ever replace data points by NaNs. So one might change this to
replaceData(handle, id_replace), or even more oscure
replaceDataWithNaN(handle, id_replace)

The current function is more flexible and can potentially be used sometimes later when the need arises.